model/testing/src/icon4py/model/testing/helpers.py

```diff
@@ -66,6 +71,33 @@ def allocate_data(backend, input_data):
     return input_data
 
 
+def match_marker(markers: tuple, param: typing.Any):
+    for marker in markers:
+        if hasattr(marker, "markname"):
+            m_name = marker.markname
+        elif hasattr(marker, "name"):
+            m_name = marker.name
+        else:
+            m_name = marker
+
+        match m_name:
+            case "embedded_remap_error" if is_embedded(param):
+                # https://github.com/GridTools/gt4py/issues/1583
+                pytest.xfail("Embedded backend currently fails in remap function.")
+            case "uses_as_offset" if is_embedded(param):
+                pytest.xfail("Embedded backend does not support as_offset.")
+            case "requires_concat_where" if is_embedded(param):
+                pytest.xfail(
+                    "Embedded backend does not support larger boundaries than field sizes."
```

Comment:
I'd propose to either just repeat what the marker says, something like
```suggestion
                    "Stencil requires concat_where."
```
or if you want to give slightly more detail something like
```suggestion
                    "Stencil needs to use concat_where, but is implemented with an outdated interpretation of `where`."
```